### PR TITLE
Android support and related minor fixes

### PR DESCRIPTION
**Pull request by hasselmm**

### CODE REVIEW

File: androidkeystore.cpp [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    return result;
+}
+
+jbyteArray toArray(const QByteArray &bytes)
+{
+    QAndroidJniEnvironment env;
+    const int length = bytes.length();
+    jbyteArray array = env->NewByteArray(length);
+    env->SetByteArrayRegion(array, 0, length, reinterpret_cast<const jbyte *>(bytes.constData()));
+    return array;
```
Comment (bog-dan-ro) [inline androidkeystore.cpp]: AFAIK NewByteArray holds a local reference to that object, so, IMHO it safer to return a QAndroidJniObject::fromLocalRef(array) instead

**Comment by hasselmm** [comment]
Split the commits, the android part is in #88 now.

File: androidkeystore.cpp (hunk 1 above)
Comment (hasselmm) [inline androidkeystore.cpp]: Ah, Qt 5.7 material. Good to know.

**Comment by frankosterfeld** [comment]
I think this can be closed now, both iOS and Android support is merged.

**Pull request closed without merging**